### PR TITLE
expo-bluetooth-utils

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

This library is used with `expo-bluetooth` to convert read/write data.

# How

* Moved `expo-bluetooth-utils` from #3131 to simplify the BLE PR
* Created expo-bluetooth-utils
* Added newest bluetooth spec for parsing Services, Characteristics, Descriptors
* Added utils for converting data to and from base64 for read/write operations.

# Test Plan

TBD

### CODE REVIEW

File: packages/expo-bluetooth-utils/src/Bluetooth.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export { default as Descriptors } from './Descriptors';
+export { default as Services } from './Services';
+
+/*
+ * Convert a JS value before sending it to native.
+ * 1. Use encodeURIComponent to get percent-encoded UTF-8 string value.
+ * 2. Convert the percent-encodings to raw bytes.
+ * 3. Pass value to Base64.btoa (Binary to ASCII)
+ */
+export function JSONToNative(str: string): string {
```
Comment (ide) [inline packages/expo-bluetooth-utils/src/Bluetooth.ts]: The "JSON" part here is confusing - there's no JSON at all -- name it something like `encodeBinaryString`

File: packages/expo-bluetooth-utils/src/Bluetooth.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  return Base64.btoa(input);
+}
+
+/*
+ * Take a native value and convert it to a JS value.
+ * 1. Create binary value from input string
+ * 2. Convert the raw bytes to percent-encodings.
+ * 3. Decode the raw bytes
+ */
+export function nativeToJSON(str: string): string {
```
Comment (ide) [inline packages/expo-bluetooth-utils/src/Bluetooth.ts]: decodeBinaryString

**Comment by Davidzhu001** [comment]
hey guys, is this still in progress?

**Comment by brentvatne** [comment]
> hey guys, is this still in progress?

yes, just temporarily working on some other stuff instead

**Comment by Davidzhu001** [comment]
@brentvatne do you have a time in mind when it will be released?

**Comment by brentvatne** [comment]
sorry no estimate at the moment, if your need is urgent you might want to look into another lib

**Comment by Davidzhu001** [comment]
@EvanBacon @brentvatne 
thanks for working on this. so appreciate!

File: packages/expo-bluetooth-utils/build/Bluetooth.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { Base64 } from 'js-base64';
```
Comment (ide) [inline packages/expo-bluetooth-utils/build/Bluetooth.js]: Would the need for this go away with Turbomodules? (Just thinking about a future version of the library..)

File: yarn.lock [hunk 4]
```diff
@@ -11911,6 +11916,11 @@ process@^0.11.1, process@^0.11.10:
   resolved "https://registry.yarnpkg.com/process/-/process-0.11.10.tgz#7332300e840161bda3e69a1d1d91a7d4bc16f182"
   integrity sha1-czIwDoQBYb2j5podHZGn1LwW8YI=
 
+process@~0.5.1:
```
Comment (ide) [inline yarn.lock]: Remove?

**Review by ide — APPROVED** [review]

**Pull request closed without merging**